### PR TITLE
Update arrow 49.0.0 and object_store 0.8.0

### LINKED ISSUE #8092: Parquet error: Invalid offset in sparse column chunk data
### Describe the bug

Querying on a registered parquet table sometimes throws an error. The query is doing a range query over multiple fields. The same query with different literals, in which only one range impacts the output as the others are unbound works fine. Also, this happens for SQL and DataFrame alike.

### To Reproduce

_No response_

### Expected behavior

Query returns result.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by tustvold**
## Which issue does this PR close?

Closes #8092

### CODE REVIEW

File: datafusion/core/tests/fifo.rs [hunk 1]
```diff
@@ -336,6 +336,7 @@ mod unix_test {
 
     /// It tests the INSERT INTO functionality.
     #[tokio::test]
+    #[ignore]
```
Comment (tustvold) [inline datafusion/core/tests/fifo.rs:339]: #8021 shows how these tests might be ported

**Comment by alamb** [comment]
I believe this upgrade is blocked on sorting out how to upgrade the design to handle FIFO files as described in   https://github.com/apache/arrow-datafusion/pull/8021

**Comment by alamb** [comment]
> Closes https://github.com/apache/arrow-datafusion/issues/8092

Technically speaking this bug was fixed by the update to 48.0.1

File: datafusion/core/src/catalog/listing_schema.rs [hunk 2]
```diff
@@ -92,12 +92,7 @@ impl ListingSchemaProvider {
 
     /// Reload table information from ObjectStore
     pub async fn refresh(&self, state: &SessionState) -> datafusion_common::Result<()> {
-        let entries: Vec<_> = self
-            .store
-            .list(Some(&self.path))
-            .await?
-            .try_collect()
-            .await?;
+        let entries: Vec<_> = self.store.list(Some(&self.path)).try_collect().await?;
```
Comment (alamb) [inline datafusion/core/src/catalog/listing_schema.rs:95]: that is certainly nicer

File: datafusion/physical-expr/src/expressions/cast.rs [hunk 3]
```diff
@@ -680,7 +680,11 @@ mod tests {
         // Ensure a useful error happens at plan time if invalid casts are used
         let schema = Schema::new(vec![Field::new("a", DataType::Int32, false)]);
 
-        let result = cast(col("a", &schema).unwrap(), &schema, DataType::LargeBinary);
+        let result = cast(
```
Comment (alamb) [inline datafusion/physical-expr/src/expressions/cast.rs:683]: Is this change needed because arrow now supports  casting Int32 --> LargeBinary?

**Review by alamb — APPROVED** [review]
Thank you @tustvold -- I know this was a long time in the making but the result is great.

I think this upgrade makes the code look much nicer (removing the double layer of errors from the object store APIs is quite nice 👨‍🍳 👌 )
